subiquity/models/filesystem.py

```diff
@@ -708,6 +708,17 @@ def ok_for_raid(self):
             return False
         return True
 
+    @property
+    def path(self):
+        return self.device.path + str(self.number)
```

Comment:
unfortunately this is too simple, it's probably ok for desktop but it won't work for RAIDs (a partition of /dev/md0 is something like /dev/md0p1) or multipath disks (where it all gets complicated). I'm probably happy with doing this this way purely for the purpose of os-prober look ups but not in an API that looks generally useful :)

It's all a bit frustrating of course because we know exactly the DEVPATH that corresponds to this partition object when making the partition object in curtin.storage_config but I don't think we attach that in any reliable way. Maybe we should... (but probably not for this branch!)

Reply:
See the fixup commit for my modification, and yes having curtin output the path sounds useful!